src/test/java/com/amazon/opendistroforelasticsearch/sql/intgtest/NestedFieldQueryTest.java

```diff
@@ -341,6 +355,37 @@ public void groupByNestedAndRegularField() {
         return kv("myNum", is(value));
     }
 
+    private Matcher<SearchHit> myNum(int[] values) {
+
+        return new BaseMatcher<SearchHit>() {
+
+            @Override
+            public boolean matches(Object item) {
+
+                if (item instanceof SearchHit) {
+                    final SearchHit hit = (SearchHit) item;
+                    List<Integer> actualValues = (ArrayList<Integer>) hit.getSourceAsMap().get("myNum");
+
+                    if (actualValues.size() != values.length) {
```

Comment:
Should we validate if the values is null?

Reply:
ideally yes, but since this is to-be-deprecated test code, and NPE will fail the test anyway, I don't see a need for that.